src/bfx_tools/pyensembl.ml

```diff
@@ -9,9 +22,12 @@ let cache_genome ~(run_with: Machine.t) ~reference_build =
   let genome = Machine.(get_reference_genome run_with reference_build) in
   let ensembl_release = genome |> Reference_genome.ensembl in
   let species = genome |> Reference_genome.species in
+  let witness_file_path = 
+    sprintf "${%s}/%s.cached" pyensembl_env_key reference_build
```

Comment:
You cannot use environment variables there; this path is not interpreted by a shell.
It should be constructed with `Machine.(get_pyensembl_cache_dir run_with)`.